Rewrite in Clojure/Script

ReilySiegel opened the pull request:
I have re-written the site in Clojure/Script to address a variety of problems with the existing site:

* Efficiency
* Maintainability
* Flexibility
* Ease of Contribution

## Changes Regarding Efficiency
The Clojure/Script version of the site is written as a react web app. While on its surface this might seem a blow to efficiency, testing reveals significantly faster FMP times due to ClojureScript's dead code elimination.

## Changes Regarding Maintainability
While the previous version of the site did not have much structure, in this version of the site each section of the site has its own namespaced file. In addition, sections of the site that are expected to change regularly are defined using pure data structures, making the site easier to update.

## Changes Regarding Flexibility
As the site is now a react app, it is unlikely that the need for a feature will arise that this technology stack is unable to accommodate.

## Changes Regarding Ease of Contribution
In addition to the benefits for maintainability, newsletter entries can be written in markdown format. A GitHub Action converts all markdown entries into HTML that is loaded into the site at compile time.

ReilySiegel committed:
Blank slate.

ReilySiegel committed:
Add newsletter builder.

The newsletter builder expects news entries to be organized in any directory
structure in the `newsletter` directory. The newsletters MUST be Markdown files,
and SHOULD contain the headers title, author, position, and date. The date
should be formatted as a string in the "yyyy-mm-dd" format.

The newsletter builder can be called with the command
`clj -A:newsletter-builder`.

ReilySiegel committed:
Add a README.org

Contains documentation for newsletter entry.

ReilySiegel committed:
Add build files to .gitignore

ReilySiegel committed:
Allow empty build directory.

ReilySiegel committed:
Add deploy to gh-pages action

ReilySiegel committed:
Fix workflow error.

ReilySiegel committed:
Remove actions/checkout@latest

ReilySiegel committed:
Fix setup-java version.

ReilySiegel committed:
Update deploy action.

ReilySiegel committed:
Update build gitignore

ReilySiegel committed:
Force orphan on gh-pages

ReilySiegel committed:
Use resources dir instead of build.

ReilySiegel committed:
Tests github pages.

ReilySiegel committed:
Re-create static portions of site.

ReilySiegel committed:
Add SASS styles

ReilySiegel committed:
Render most recent newsletter entry.

ReilySiegel committed:
Render six most recent newsletters.

ReilySiegel committed:
Enable persistent CNAME

ReilySiegel committed:
Optimizations

ReilySiegel committed:
Add images

ReilySiegel committed:
Add Rush section and smooth scrolling.

ReilySiegel committed:
Align hero image top.

ReilySiegel committed:
Add philanthropy links.

ReilySiegel committed:
Add officers

ReilySiegel committed:
Fix issue with hero links.

ReilySiegel committed:
Add footer.

ReilySiegel committed:
Remove old philanthropic organizations.

ReilySiegel committed:
Add more detailed documentation for the process of submitting a newsletter entry.

ReilySiegel committed:
Update build.yaml to use www subdomain.

Perezjo94 commented on CNAME:
```diff
@@ -1 +0,0 @@
-axpdsp.org
```
Are we moving hosting out of github pages / will keep this domain?

ReilySiegel replied:
We are keeping the hosting on github pages. The CNAME file is automatically generated in the gh-pages branch by the build action. I was planning on keeping the existing domain name, that will need to be changed in build.yaml after it is merged.

Perezjo94 replied:
gotcha 👍

ReilySiegel committed:
Add a slideshow of brother "profiles" based on a Google Sheet.

ReilySiegel commented:
At the request of PR Committee, I have updated the website to include a section for brother "profiles", with the data collected from a [Google Sheet](https://docs.google.com/spreadsheets/d/1o7fckgforDgA_YYQkBFjZtVr7Z2U19DJsnwG5MrMEJE/edit?usp=sharing). Feel free to suggest any changes. Changing the spreadsheet will update the site on page reload. Again, a live version of the site can be found at https://crows.house

ReilySiegel committed:
Store Officer data in Brother spreadsheet.

ReilySiegel committed:
Force Brotherhood section fullheight and vcentered

ReilySiegel committed:
Expand information shown in brother bio.

ReilySiegel committed:
Remove custom domain until we merge to existing repo.

ReilySiegel committed:
Remove sample newsletter entry

ReilySiegel committed:
Add an `important` metatdata option to newsletter entries.

This option will allow an "important" newsletter entry to be displayed first,
regardless of its date.

ReilySiegel committed:
Improve the handling of long newsletters

Show newsletters full-width on all devices, and hide all newsletters initially.

ReilySiegel committed:
Update COVID-19 newsletter

ReilySiegel committed:
Temporarily replace Brother Bios with a placeholder image.

ReilySiegel committed:
Add CNAME

Perezjo94 reviewed (approved):
💯
